pydra/engine/workers.py

```diff
@@ -246,27 +209,60 @@ def run_el(self, runnable, rerun=False):
         script_dir, batch_script = self._prepare_runscripts(runnable, rerun=rerun)
         if (script_dir / script_dir.parts[1]) == gettempdir():
             logger.warning("Temporary directories may not be shared across computers")
-
         if isinstance(runnable, TaskBase):
-            checksum = runnable.checksum
             cache_dir = runnable.cache_dir
             name = runnable.name
+            uid = runnable.uid
         else:
-            checksum = runnable[-1].checksum_states()[runnable[0]]
```

Comment:
i'm not following this logic here, perhaps useful to add to docstring. how long can runnable be? what is the 0 index and the -1 index?

Reply:
i've added a comment, this is for situation when `runnable` is in fact (ind, pkl file, runnable)